### ISSUE TITLE
Rate Limiting by header doesn't supported ?

### ISSUE THREAD

**Issue by mholleville** [Kong/kubernetes-ingress-controller#779]
Hi ! 

I saw on the plugin hub which we can rate limiting by header_name : https://docs.konghq.com/hub/kong-inc/rate-limiting/

When I'm trying to deploy on Kubernetes I have the following message. 

```
400 Bad Request {"fields":{"plugins":[{"config":{"header_name":"unknown field","limit_by":"expected one of: consumer, credential, ip, service"}}]},"name":"invalid declarative configuration","code":14,"message":"declarative config is invalid: {plugins={{config={header_name=\"unknown field\",limit_by=\"expected one of: consumer, credential, ip, service\"}}}}"}
```

With the ingress controller we can only rate limit by `consumer`, `ip`, `service`

My Deployment file 

```
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: global-rate-limit
  labels:
    global: "true"
service: Test
config:
  minute: 2
  header_name: Custom-header
  policy: local
  limit_by: header
plugin: rate-limiting
```
Can you tell me if the rate limiting by header_name was possible which another way ?

**Comment by ashnigam** [comment]
I also ran into same issue yesterday.

**Comment by rainest** [comment]
On which version? That option was not available prior to 2.1: https://github.com/Kong/kong/commit/7c449842a55fc0d5d9077bd5b6312bedf2edd450

We don't perform any additional validation of that setting in the controller, so it's only rejected if the Kong admin API rejects it.

The Docs team is working on some changes that will make it clearer when features are not available on older versions, but for the time being be aware that plugin documentation generally reflects the options available in the latest release only.

Will go ahead and close this on the assumption that it's most likely just that version gap, but if you see the same issue when using Kong 2.1, please file a report in https://github.com/Kong/kong
